pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -124,12 +130,45 @@ func createRawDeployment(componentMeta metav1.ObjectMeta, workerComponentMeta me
 	return deploymentList
 }
 
-func createRawDefaultDeployment(componentMeta metav1.ObjectMeta,
+func createRawDefaultDeployment(clientset kubernetes.Interface, componentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec,
 	podSpec *corev1.PodSpec) *appsv1.Deployment {
 	podMetadata := componentMeta
 	podMetadata.Labels["app"] = constants.GetRawServiceLabel(componentMeta.Name)
 	setDefaultPodSpec(podSpec)
+	var isvcname string
+	if val, ok := componentMeta.Labels[constants.InferenceServiceLabel]; ok {
+		isvcname = val
+	} else {
+		isvcname = componentMeta.Name
+	}
+	var upstreamPort string
+	if val, ok := componentMeta.Labels[constants.ODHKserveRawAuth]; ok && val == "true" {
+		if componentExt != nil && componentExt.Batcher != nil {
+			upstreamPort = constants.InferenceServiceDefaultAgentPortStr
+		} else if componentExt != nil && componentExt.Logger != nil {
+			upstreamPort = constants.InferenceServiceDefaultAgentPortStr
+		} else {
+			upstreamPort = GetKServeContainerPort(podSpec)
+			if upstreamPort == "" {
+				upstreamPort = constants.InferenceServiceDefaultHttpPort
+			}
+		}
+		oauthProxyContainer, err := generateOauthProxyContainer(clientset, isvcname, componentMeta.Namespace, upstreamPort)
+		if err == nil {
+			podSpec.Containers = append(podSpec.Containers, oauthProxyContainer)
+		}
+		tlsSecretVolume := corev1.Volume{
+			Name: tlsVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  componentMeta.Name + constants.ServingCertSecretSuffix,
```

Comment:
not sure if I understand the purpose of this tlsSecret here, where and how will it be used? And, who will be creating the secret?